common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerEncryptionTests.java

```diff
@@ -0,0 +1,101 @@
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.platform;
+
+import android.os.Build;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import com.microsoft.identity.common.exception.ClientException;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.microsoft.identity.common.internal.platform.IDevicePopManager.Cipher.RSA_ECB_OAEPWithSHA_256AndMGF1Padding;
+import static com.microsoft.identity.common.internal.platform.IDevicePopManager.Cipher.RSA_ECB_OAEPWithSHA_384AndMGF1Padding;
+import static com.microsoft.identity.common.internal.platform.IDevicePopManager.Cipher.RSA_ECB_OAEPWithSHA_512AndMGF1Padding;
+import static com.microsoft.identity.common.internal.platform.IDevicePopManager.Cipher.RSA_ECB_PKCS1_PADDING;
+
+// Note: Test cannot use robolectric due to the following open issue
+// https://github.com/robolectric/robolectric/issues/1518
+@RunWith(Parameterized.class)
+public class DevicePoPManagerEncryptionTests {
+
+    private static final String DATA_TO_ENCRYPT = "The quick brown fox jumped over the lazy dog.";
+
+    private final IDevicePopManager devicePopManager;
+    private final IDevicePopManager.Cipher cipher;
+
+    @Parameterized.Parameters
+    public static Iterable<IDevicePopManager.Cipher> testParams() {
+        final List<IDevicePopManager.Cipher> ciphers = new ArrayList<>();
+
+        // Only execute these tests at appropriate API levels...
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            ciphers.add(RSA_ECB_PKCS1_PADDING);
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            ciphers.add(RSA_ECB_OAEPWithSHA_256AndMGF1Padding);
+            ciphers.add(RSA_ECB_OAEPWithSHA_384AndMGF1Padding);
+            ciphers.add(RSA_ECB_OAEPWithSHA_512AndMGF1Padding);
+        }
+
+        return ciphers;
+    }
+
+    public DevicePoPManagerEncryptionTests(final IDevicePopManager.Cipher cipher)
+            throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        devicePopManager = new DevicePopManager();
+        this.cipher = cipher;
+    }
+
+    @Before
+    public void setUp() throws ClientException {
+        devicePopManager.generateAsymmetricKey(ApplicationProvider.getApplicationContext());
+    }
+
+    @After
+    public void tearDown() {
+        devicePopManager.clearAsymmetricKey();
+    }
+
+    @Test
+    public void testEncryption() throws ClientException {
+        final String cipherText = devicePopManager.encrypt(cipher, DATA_TO_ENCRYPT);
```

Comment:
For grins, here, maybe make sure that !DATA_TO_ENCRYPT.equals(cipherText)

Reply:
ACK

Reply:
4ec108d2f1cec690343eee0713c55b9a4afe2a9d